### DISCUSSION TITLE
GUI: Add more tests after refactoring

### DISCUSSION THREAD

**Discussion by chenghuzi** [jonescompneurolab/hnn-core#493] [GSoC]
Just added a simple test for our gui bf2831b9a981eac808e8744b58b1582b09814de7. I think probably we need to add more tests for those callback functions only after refactoring the current function-based code into object-base one. As callbacks are operating on certain ipywidgets, if we have an object which gives us access to those widget members it will be much easier for testing. We can do this after merging the current PR.

**Comment by jasmainak** [comment]
Fair enough. Can you start sketching out the object? What attributes will it contain and what methods?

**Comment by jasmainak** [comment]
I am thinking we should make a class that looks like this:

```py
gui = HNNGUI()
gui.load_network(fname)
gui.add_drive('proximal', 'evoked')
gui.widgets['drives']['evoked0']['time'] = 5.2
gui.run()
```

If we design it this way, it will be straightforward to test and also to document. Do you want to give it a shot?

**Reply by chenghuzi** [reply]
Agreed, I'm experimenting now and probably will open a new PR soon
